src/Symfony/Component/Form/ChoiceList/Factory/CachingFactoryDecorator.php

```diff
@@ -150,6 +150,38 @@ public function createListFromLoader(ChoiceLoaderInterface $loader, $value = nul
     }
 
     /**
+     * Filters a loaded choice list.
+     *
+     * The callable should be compatible with array_filter and
+     * return true to keep a loaded choice or false to discard it.
+     *
+     * @param ChoiceListInterface $choiceList The loaded choice list to filter
+     * @param callable            $filter     The filter to apply to the choices
+     *
+     * @return ChoiceListInterface The filtered choice list
+     */
+    public function filterChoiceList(ChoiceListInterface $choiceList, callable $filter)
```

Comment:
Adding this method to this class is wrong, as this class is only a particular implementation of ChoiceListFactoryInterface and ChoiceType must accept any implementation